R/intervals_support_funs.R

```diff
@@ -0,0 +1,244 @@
+#'   time = rep(0:5, 2),
+#'   analyte = rep(c("Analyte1", "Analyte2"), each = 6),
+#'   include_hl = c(FALSE, NA, TRUE, TRUE, TRUE, TRUE, FALSE, NA, TRUE, TRUE, TRUE, TRUE)
+#' )
+#'
+#' d_dose <- data.frame(
+#'   dose = c(100, 200),
+#'   time = c(0, 0),
+#'   treatment = c("A", "B"),
+#'   ID = c(1, 2)
+#' )
+#'
+#' o_conc <- PKNCAconc(d_conc, conc ~ time | analyte, include_half.life = "include_hl")
+#' o_dose <- PKNCAdose(d_dose, dose ~ time | treatment + ID)
+#'
+#' intervals <- data.frame(
+#'   start = c(0, 0, 0),
+#'   end = c(24, 48, Inf),
+#'   half.life = c(TRUE, FALSE, TRUE),
+#'   impute = c("start_conc0,start_predose", "start_predose", "start_conc0"),
+#'   ANALYTE = c("Analyte1", "Analyte2", "Analyte1"),
+#'   ROUTE = c("intravascular", "oral", "intravascular")
+#' )
+#'
+#' o_data <- PKNCAdata(o_conc, o_dose, intervals = intervals)
+#'
+#' # Apply interval_remove_impute function
+#' o_data <- interval_remove_impute(o_data, target_impute = "start_conc0", target_params = c("half.life"), target_groups = list(ANALYTE = "Analyte1", ROUTE = "intravascular"))
+#'
+#' # Print updated intervals
+#' print("Updated intervals:")
+#' print(o_data$intervals)
+#'
+#' @export
+interval_remove_impute <- function(data, target_impute, target_params = NULL, target_groups = NULL) {
+  # Validate the input
+  if (missing(data) || missing(target_impute)) {
+    stop("Both 'data' and 'target_impute' must be provided.")
+  }
+
+  if (!("intervals" %in% names(data)) || !("PKNCAdata" %in% class(data))) {
+    stop("'data' object must be a PKNCAdata object with 'intervals' and 'data' components.")
+  }
+
+  if (!is.character(target_impute)) {
+    stop("'target_impute' must be a character string.")
+  }
+
+  # Get all parameter column names in the PKNCAdata object
+  all_param_options <- names(sapply(PKNCA.options()$single.dose.aucs, is.logical))
+  logical_cols <- names(which(colSums(data$intervals[sapply(data$intervals, is.logical)]) > 1))
+  param_cols <- intersect(logical_cols, all_param_options)
+
+  # Handle target_params
+  if (is.null(target_params)) {
+    # Take all logical columns in data$intervals that are known parameters
+    target_params <- param_cols
+  } else {
+    # Check that all target_params are logical columns in data$intervals and known parameters
+    missing_params <- setdiff(target_params, param_cols)
+    if (length(missing_params) > 0) {
+      stop("The following target_params are not interval columns and/or known PKNCA parameters: ", paste(missing_params, collapse = ", "))
+      target_params <- intersect(target_params, param_cols)
+    }
+  }
+
+  # Determine the name of the impute column
+  impute_col <- if (!is.na(data$impute)) {
+    data$impute
+  } else if ("impute" %in% colnames(data$intervals)) {
+    "impute"
+  } else {
+    stop("The 'data$intervals' object must contain an impute column either defined in the PKNCAdata object or called `impute`.")
```

Comment:
Please make this a warning instead of an error. If the goal is to remove an imputation method and there is no imputation, then the goal is achieved.